common/util.go

```diff
@@ -530,3 +538,130 @@ func ParseEthAddr(strJsonKey string) (string, error) {
 	}
 	return "", errors.New("Error parsing address from keyfile")
 }
+
+// determines the duration of an mp3 audio file by reading the frames
+var ErrUnsupportedFormat = errors.New("Unsupported audio file format. Supported formats: mp3, wav, mp4, m4a, webm, flac")
+var ErrorCalculatingDuration = errors.New("Error calculating duration")
+
+func CalculateAudioDuration(audio types.File) (int64, error) {
+	read, err := audio.Reader()
+	if err != nil {
+		return 0, err
+	}
+	defer read.Close()
+
+	fileExt := filepath.Ext(audio.Filename())
+	switch fileExt {
+	case ".wav":
+		return GetDuration_WAV(read)
+	case ".mp4":
+		return GetDuration_MP4(read)
+	case ".m4a":
+		return GetDuration_MP4(read)
+	case ".webm":
+		return GetDuration_WEBM(read)
+	case ".flac":
+		return GetDuration_FLAC(read)
+	case ".mp3":
+		return GetDuration_MP3(read)
+
+	default:
+		return 0, ErrUnsupportedFormat
+	}
+}
+
+func GetDuration_WAV(audio io.ReadCloser) (int64, error) {
```

Comment:
I think that all these functions would benefit from unit tests.

Reply:
Removed functions, writing tests is in the backlog